### ISSUE TITLE
Is the yyyy of CalVer the correct level for run-export?

### ISSUE THREAD

**Issue by h-vetinari** [conda-forge/intel_repack-feedstock#81]
MKL currently does
https://github.com/conda-forge/intel_repack-feedstock/blob/c007b4da38dfd8056ff59b6a7a089507ed2c16ce/recipe/meta.yaml#L453-L460

which uses the implicit `max_pin='x'` major-level pinning for `pin_subpackage`. This would mean MKL 2023.{0, 1, 2} are ABI-compatible between each other, but not with 2024.x (which are themselves again inter-compatible).

This sounds fishy to me, at least I'm not aware of the year-portion of the CalVer indicating compatibility here.

Could you please enlighten us here @conda-forge/intel_repack?

CC @hmaarrfk, this impacts the mkl2024 migration, which might or might not be useful in light of this.

**Comment by napetrov** [comment]
@h-vetinari  - sorry for late response here. In short Intel use strange versioning, but practically this is semantic versioning and first number being major version, so it would indicate ABI/API breakage, while minor versions preserve compatibility. 

There are interesting results of that policy - if there is no breakage at a time of oneAPI release we would keep incrementing minor versions, so versions and actual release years wouldn't match. This likely would be fixed by larger increments but we might endup in 
2025.0 -> 2025.1 -> 2025.2 -> 2025.3 -> 2025.4 -> 2025.5 -> 2027.0 versioning.  

But from ABI perspective we can assume that semantic versioning would be followed and only major version updates would be introducing breaking changes.

**Comment by h-vetinari** [comment]
OK, thanks for elaborating on this. So far, the yyyy-portion seems to have been closely aligned with actual calendar years, so the part with the ABI/API stability an `... -> 2025.5 -> 2027.0` was really not apparent. But fine by me if that's the case. 👍

**Comment by napetrov** [comment]
Yes, and i hope we would keep it aligned as some cleanup and breaking changes should happen.  But technically  it's wired mix at this point.  But reliable from ABI perspective
